lib/controllers/login.js

```diff
@@ -33,10 +33,24 @@ module.exports = function(req, res) {
       options = {};
     }
 
+    var hasSocialProviders = false;
+
+    // Iterate all social providers and check if there are any enabled.
+    if (config.socialProviders) {
+      for (var name in config.socialProviders) {
+        var provider = config.socialProviders[name];
+        if (provider.enabled) {
+          hasSocialProviders = true;
+          break;
+        }
+      }
+    }
+
```

Comment:
Thanks! I definitely agree that there are definitely nicer solutions:) I'm not that much for iterating over an object using Object.keys like that. But if you're familiar with lodash they have pretty nice helper methods for collections such as this. I.e:

```js
var hasSocialProviders = _.any(config.socialProviders, function (socialProvider) {
  return socialProvider.enabled;
});
```

I'll let you know when I have updated this.